### PR TITLE
Add type and angle picking to the custom gradient component

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR adds a UI to pick choose between linear and radial gradients and a UI to pick an angle on linear gradients.
The UI is based on some of the designs suggested in https://github.com/WordPress/gutenberg/issues/16662.

The changes seem huge but the file custom-gradient-bar.js is purely a rename from index.js with some class name changes plugs a logic to support radial gradient in the bar. I made the rename in a separate commit but GitHub continues to show all the file as new :(
The file index.js is new (while GitHub shows lots of removals from the old file.

## How has this been tested?
I verified I can choose between radial and linear gradients.
I verified for linear gradients I can choose an angle for the gradient.
## Screenshots 
Jan-11-2020 12-28-00

Jan-11-2020 12-25-25

### CODE REVIEW

**Comment by jorgefilipecosta** [comment]
This PR was rebased and now includes the angle picker by default. It is ready for a review.

**Comment by mapk** [comment]
I'm excited to get this in! Here's some feedback that might help tighten this up a bit more.

1. Let’s minimize those tabs (Solid color | Gradient) as much as possible. They’re too big right now.
2. Change the word "Overlay" to "Color".
3. Remove the border around the gradient type icons.
4. Provide more vertical padding between all the parts. (24px or some other CSS variable close to that)
5. Make sure the Angle number field and the "Clear" button align along the right side nicely.

These changes should look something like this:

Screen Shot 2020-02-06 at 12 03 38 PM

**Comment by mapk** [comment]
One more minor text change here that could help.

Accordion title = `Overlay`
Label next to swatch = `Color`
Tabs = `Solid | Gradient`

This may help avoid repetition in the text.

File: packages/components/src/custom-gradient-picker/utils.js [hunk 1]
```diff
@@ -174,3 +178,33 @@ export function getMarkerPoints( gradientAST ) {
...
+		type: 'linear-gradient',
+		orientation: HORIZONTAL_GRADIENT_ORIENTATION,
+		colorStops: gradientAST.colorStops,
+	} );
+}
+
+export function getGradientParsed( value ) {
+	let hasGradient = !! value;
+	// gradientAST will contain the gradient AST as parsed by gradient-parser npm module.
+	// More information of its structure available at.
```
Comment (draganescu) [inline packages/components/src/custom-gradient-picker/utils.js]: > More information of its structure available at:
I think we're missing a link in this comment.

Comment (jorgefilipecosta) [reply packages/components/src/custom-gradient-picker/utils.js]: Nice catch 👍

**Review by draganescu — APPROVED** [review]
Tested and works perfectly. Codewise everything looks good.

**Comment by ItsJonQ** [comment]
@mapk Hallo!! I'm helping out @jorgefilipecosta with the design follow ups :)

Will create a separate PR with updates!

**Comment by ItsJonQ** [comment]
@mapk Haii! I had a question 👋 

> Remove the border around the gradient type icons.

If we do this, then there's no indication of which gradient type is active:

Screen Shot 2020-02-07 at 11 06 13 AM

How would you like to proceed?

**Comment by mapk** [comment]
> How would you like to proceed?

The grey border in that image should be removed. But the grey background on the icon when it's selected should remain. I hope that helps!

**Comment by karmatosed** [comment]
I wonder if a simple square and either circle in it or arrow would work. I really like what Figma does and riffed on that in the monotone (Figma screenshots in green).

Frame 1

I added on states to these to show.

**Comment by mapk** [comment]
Oooh! I really like these two:

Screen Shot 2020-02-07 at 9 01 15 AM

Let's go with these. I'd like to see these in relation to our already standard icon selection background indicator (dark grey background). If that doesn't work too well, we can look into the thinner selection idea.

**Pull request merged**